usr.bin/sockstat/sockstat.c

```diff
@@ -1414,69 +1434,75 @@ display_sock(struct sock *s, struct col_widths *cw, char *buf, size_t bufsize)
 					(s->proto == IPPROTO_TCP &&
 					s->state != TCPS_CLOSED &&
 					s->state != TCPS_LISTEN))) {
-				printf(" %*u", cw->encaps,
+				xo_emit(" {:encaps/%*u}", cw->encaps,
 					ntohs(faddr->encaps_port));
 			} else
-				printf(" %*s", cw->encaps, "??");
+				xo_emit(" {:encaps/%*s}", cw->encaps, "??");
 		}
 		if (opt_s) {
 			if (faddr != NULL &&
 				s->proto == IPPROTO_SCTP &&
 				s->state != SCTP_CLOSED &&
 				s->state != SCTP_BOUND &&
 				s->state != SCTP_LISTEN) {
-				printf(" %-*s", cw->path_state,
+				xo_emit(" {:path-state/%-*s}", cw->path_state,
 					sctp_path_state(faddr->state));
 			} else
-				printf(" %-*s", cw->path_state, "??");
+				xo_emit(" {:path-state/%-*s}",
+					cw->path_state, "??");
 		}
 		if (first) {
 			if (opt_s) {
 				if (s->proto == IPPROTO_SCTP ||
 				    s->proto == IPPROTO_TCP) {
 					switch (s->proto) {
 					case IPPROTO_SCTP:
-						printf(" %-*s", cw->conn_state,
-						    sctp_conn_state(s->state));
+						xo_emit(" {:path-state/%-*s}",
+							cw->path_state,
+							sctp_path_state(
+								faddr->state));
 						break;
 					case IPPROTO_TCP:
 						if (s->state >= 0 &&
 							s->state < TCP_NSTATES)
-							printf(" %-*s",
+						    xo_emit(" {:conn-state/%-*s}",
 							cw->conn_state,
 							tcpstates[s->state]);
 						else
-							printf(" %-*s",
+						    xo_emit(" {:conn-state/%-*s}",
 							cw->conn_state, "??");
 						break;
 					}
 				} else
-					printf(" %-*s", cw->conn_state, "??");
+					xo_emit(" {:conn-state/%-*s}",
+						cw->conn_state, "??");
 			}
 			if (opt_S) {
 				if (s->proto == IPPROTO_TCP)
-					printf(" %-*s", cw->stack, s->stack);
+					xo_emit(" {:stack/%-*s}",
+						cw->stack, s->stack);
 				else
-					printf(" %-*s", cw->stack, "??");
+					xo_emit(" {:stack/%-*s}",
+						cw->stack, "??");
 			}
 			if (opt_C) {
 				if (s->proto == IPPROTO_TCP)
-					printf(" %-*s", cw->cc, s->cc);
+					xo_emit(" {:cc/%-*s}", cw->cc, s->cc);
 				else
-					printf(" %-*s", cw->cc, "??");
+					xo_emit(" {:cc/%-*s}", cw->cc, "??");
 			}
 		}
 		if (laddr != NULL)
 			laddr = laddr->next;
 		if (faddr != NULL)
 			faddr = faddr->next;
 		if (laddr != NULL || faddr != NULL)
-			printf("%-*s %-*s %-*s %-*s %-*s", cw->user, "",
-				cw->command, "", cw->pid, "", cw->fd, "",
-				cw->proto, "");
+			xo_emit("{:uid/%-*s} {:command/%-*s} {:pid/%*s} "
```

Comment:
The "??" fields are nice for human-readable output.  But for machine-readable output I don't think they make sense.  Better to simply not print those fields, I think.